### PR TITLE
Improve failure messages from output matchers

### PR DESCRIPTION
**Pull request by mvz**
## Summary

Make the message from the output matchers/steps provide enough information to figure out what went wrong.

## Details

* Make `have_output` matcher failure message print the command name and the full actual output.
* Make `include_an_object` matcher not print boiler plate when the collection has exactly one member.
* Remove documentation for `include_an_object` since it is non-public API.
* Remove obsolete RSpec version checks

## Motivation and Context

The build failures for #544 do not provide enough information to see what failed because the actual output is truncated to a single-line string.

## How Has This Been Tested?

Features and specs were added.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- I've added tests for my code
- My change requires a change to the documentation.
- I have updated the documentation accordingly.

### CODE REVIEW

File: features/03_testing_frameworks/cucumber/steps/command/check_output_of_command.feature [hunk 1]
```diff
@@ -41,6 +41,28 @@ Feature: All output of commands which were executed
...
+    """cucumber
+    Feature: Run command
+      Scenario: Run command
+        When I run `aruba-test-cli`
+        Then the output should contain "goodbye world"
+    """
+    When I run `cucumber`
+    Then the features should not all pass with:
+    """
+          expected `aruba-test-cli` to have output string includes: "goodbye world"
```
Comment (olleolleolle) [inline features/03_testing_frameworks/cucumber/steps/command/check_output_of_command.feature]: **Narrator:** _I was missing something_.

Perhaps I am missing something, but line 61 and line 56 have mismatched wording.

>  +          expected `aruba-test-cli` to have output string includes: "goodbye world"

"to have output string _including_" would be my personal expectation (just reading the line)

File: features/03_testing_frameworks/cucumber/steps/command/check_output_of_command.feature [hunk 2]
```diff
@@ -179,22 +201,32 @@ Feature: All output of commands which were executed
...
+        \"\"\"
+        hello
+        world
+        \"\"\"
     """
     When I run `cucumber`
-    Then the features should all pass
+    Then the features should not all pass with:
+    """
+          expected `aruba-test-cli` to have output output string is eq: "hello\nworld"
```
Comment (olleolleolle) [inline features/03_testing_frameworks/cucumber/steps/command/check_output_of_command.feature]: ...but now I understand there's more to it, we're surfacing RSpec matchers more directly in these messages.

File: lib/aruba/matchers/command/be_successfully_executed.rb [hunk 3]
```diff
@@ -29,6 +29,4 @@
   end
 end
 
-if RSpec::Expectations::Version::STRING >= '3.1'
-  RSpec::Matchers.define_negated_matcher :have_failed_running, :be_successfully_executed
-end
+RSpec::Matchers.define_negated_matcher :have_failed_running, :be_successfully_executed
```
Comment (olleolleolle) [inline lib/aruba/matchers/command/be_successfully_executed.rb]: Great cleanup.

File: features/03_testing_frameworks/cucumber/steps/command/check_output_of_command.feature (hunk 2 above)
Comment (olleolleolle) [reply features/03_testing_frameworks/cucumber/steps/command/check_output_of_command.feature]: Please disregard my first comment.

**Comment by mvz** [comment]
Right. I still think this is an improvement, but it didn't improve the case for negative failures of `include_an_object` so didn't help with the original problem :disappointed:.

**Pull request merged**